src/Components/CommentForm.jsx

```diff
@@ -0,0 +1,77 @@
+import { useState } from 'react';
+import { postComment } from '../api';
+
+const CommentForm = ({ articleId, onCommentPosted }) => {
+  const [comment, setComment] = useState('');
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [error, setError] = useState(null);
+  const [successMessage, setSuccessMessage] = useState('');
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    
+    if (!comment.trim()) {
+      setError('Comment cannot be empty');
+      return;
+    }
+
+    setIsSubmitting(true);
+    setError(null);
+    setSuccessMessage('');
+
+    postComment(articleId, comment.trim())
+      .then((newComment) => {
+        setComment('');
+        setSuccessMessage('Comment posted successfully');
+        if (onCommentPosted) {
+          onCommentPosted();
+        }
+        setTimeout(() => {
+          setSuccessMessage('');
+        }, 3000);
+      })
+      .catch((err) => {
```

Comment:
This catch block will never be reached, as you have one within your postComment function... there should be one catch at the end of the chain (which is here)